src/test/ui/unused/unused-field-assign.stderr

```diff
@@ -0,0 +1,119 @@
+error: unused assignement to temporary
+  --> $DIR/unused-field-assign.rs:11:20
+   |
+LL |     Foo { x: 2 }.x = 3;
+   |     ------------   ^
+   |     |
+   |     this is not bound to any variable
+   |
+note: the lint level is defined here
+  --> $DIR/unused-field-assign.rs:1:9
+   |
+LL | #![deny(unused_assignments)]
+   |         ^^^^^^^^^^^^^^^^^^
+help: consider assigning this struct to a local variable
+  --> $DIR/unused-field-assign.rs:11:5
+   |
+LL |     Foo { x: 2 }.x = 3;
+   |     ^^^^^^^^^^^^
```

Comment:
I feel that this `help` is not necessarily correct for this case (but would be for the `const`).